Neos.Fusion/Classes/FusionObjects/RendererImplementation.php

```diff
@@ -20,51 +20,45 @@
  */
 class RendererImplementation extends AbstractFusionObject
 {
+    public function getRenderer(): mixed
+    {
+        return $this->fusionValue('renderer');
+    }
 
-    /**
-     * The type to render if condition is true
-     *
-     * @return string
-     */
-    public function getType()
+    public function getRenderPath(): ?string
+    {
+        return $this->fusionValue('renderPath');
+    }
+
+    public function getType(): ?string
     {
         return $this->fusionValue('type');
     }
 
-    /**
-     * A path to a Fusion configuration
-     *
-     * @return string
-     */
-    public function getRenderPath()
+    private function canRenderWithRenderer()
     {
-        return $this->fusionValue('renderPath');
+        return $this->runtime->canRender($this->path . '/renderer');
     }
 
-    /**
-     * Render $type and return it.
-     *
-     * @return mixed
-     */
     public function evaluate()
     {
-        $rendererPath = sprintf('%s/renderer', $this->path);
-        $canRenderWithRenderer = $this->runtime->canRender($rendererPath);
-        $renderPath = $this->getRenderPath();
+        if ($this->canRenderWithRenderer()) {
+            return $this->getRenderer();
+        }
 
-        if ($canRenderWithRenderer) {
-            $renderedElement = $this->runtime->evaluate($rendererPath, $this);
-        } elseif ($renderPath !== null) {
-            if (substr($renderPath, 0, 1) === '/') {
-                $renderedElement = $this->runtime->render(substr($renderPath, 1));
-            } else {
-                $renderedElement = $this->runtime->render($this->path . '/' . str_replace('.', '/', $renderPath));
+        if ($this->getRenderPath() !== null) {
```

Comment:
`$this->getRenderPath()` is currently called between 2-3 times. I would prefer a variable here.

Reply:
i partially agree

the getter is already cached inside $this->fusionValue() so it seems odd (and sad) to save the result of a getter in a variable as this makes the code harder to read.

but as we already discussed: i want to extract the renderer logic into a trait, where we would be using no getters at all so problem solved ;)

Reply:
adjusted with 4d7d6c7615918f2edd336c9749fcb1fbe2445e6b